fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -1464,6 +1485,12 @@ public Comparison translateCorrelations(@Nonnull final TranslationMap translatio
             return new ValueComparison(type, comparandValue.translateCorrelations(translationMap), parameterRelationshipGraph);
         }
 
+        @Nonnull
+        @Override
+        public Comparison translateValue(@Nonnull UnaryOperator<Value> valueTranslator) {
+            return withValue(valueTranslator.apply(comparandValue));
+        }
+
```

Comment:
Can you make this implementation the `default` in `Comparison` and then remove ll the overrides in all the overriding comparisons?

Reply:
I am afraid that is not possible. `comparandValue` is special to `ValueComparison`. the two remaining overrides in `MultiColumnComparison` and `InvertedFunctionComaprison` are different than this one, so there is no value of moving it up, right?

Reply:
Right, but the special stuff happens in `withValue()` and its overrides so you just need to call that one.